src/seeds/types.ts

```diff
@@ -0,0 +1,107 @@
+import { Table } from '../db/db';
+import db from '../db/sql-db'
+import { getCrdtUpdateMessage, getCrdtUpsertMessage } from '../types/message'
+import { NFTContractTypeName, NFTStandard } from '../types/nft';
+import { MusicPlatformType } from '../types/platform';
+
+enum SeedEntities {
+  'platforms',
+  'nftFactories',
+  'artists',
+  'processedTracks'
+}
+type SeedEntity = keyof typeof SeedEntities;
+
+const PlatformRequiredKeys = ['id', 'name', 'type'];
+const PlatformValidKeys = PlatformRequiredKeys;
+
+const NFTFactoryRequiredKeys = ['id', 'startingBlock', 'platformId', 'contractType', 'name', 'symbol', 'typeMetadata', 'standard', 'autoApprove', 'approved'];
+const NFTFactoryValidKeys = NFTFactoryRequiredKeys;
+
+const ArtistRequiredKeys = ['id'];
+const ArtistValidKeys = ['id', 'name'];
+
+const ProcessedTrackRequiredKeys = ['id'];
+const ProcessedTrackValidKeys = ['id', 'title', 'slug', 'description', 'websiteUrl', 'lossyAudioURL', 'lossyAudioIPFSHash', 'lossyArtworkURL', 'lossyArtworkIPFSHash'];
```

Comment:
as discussed, should we remove slug for now?

also, this makes me realize that there's also an issue if we allow lossyAudioURL/lossyAudioIPFSHash to be changed (and same for artwork):
 - ideally, we expect these both to point to the same files, just one to a http url and the other to an ipfs hash. in practice at the moment this is always the case, because we upload the files to ipfs if they aren't there, but, for exapmle if we allow the audioURL to be changed after this has occurred, it will not necessarily re-upload the new track nor generate the new ipfsHash.

Maybe we should leave these 4 fields out for now too?

ie, something like:

```suggestion
const ProcessedTrackValidKeys = ['id', 'title', 'description', 'websiteUrl'];
```

Reply:
Huh, that's a great point. Some of these edge cases seem to point to the need for triggering side-effects when changing a field. e.g. 

- when a title changes, save the old slug for backward compatibility and generate a new one
- when changing artwork/audio, ensure that uploads and pin jobs are retried

It feels like it'll be well suited to the the CRDT system, or maybe an adjacent layer to it.

But yeah, in the meantime we must be conservative with what can be adjusted, will remove those keys 👍